core/src/main/scala/org/apache/spark/HttpFileServer.scala

```diff
@@ -42,10 +42,11 @@ private[spark] class HttpFileServer(
     fileDir.mkdir()
     jarDir.mkdir()
     logInfo("HTTP File server directory is " + baseDir)
-    httpServer = new HttpServer(conf, baseDir, securityManager, requestedPort, "HTTP file server")
+    httpServer = new HttpServer(conf, baseDir, securityManager, requestedPort, "HTTP file server", Some("spark.fileserver.advertisedPort"))
```

Comment:
It will revert to the regular port setting: https://github.com/apache/spark/pull/9608/files#diff-25ede712da47f7ff04ddcffce9baee45R160